src/physicsobjects.cxx

```diff
@@ -256,6 +256,72 @@ ROOT::RDF::RNode SelectedObjects(ROOT::RDF::RNode df,
                      {inputmaskname});
 }
 
+/**
+ * @brief Function used to veto a particle, if it is overlapping within a given
+ * DeltaR value with another particle within a mask. For the particle to test,
+ * the lorentz vector is used, for the veto, the mask and all input quantities
+ * needed to calculate the lorentz vector are used. Returns true if the particle
+ * is vetoed, false if it is not.
+ *
+ * @param df The input dataframe
+ * @param output_flag The name of the veto flag to be added as column to the
+ * dataframe
+ * @param p4 The name of the Lorentz vector column to be used for the particle
+ * to test
+ * @param particle_mask The name of the mask to be used for the veto
+ * @param particle_pt The name of the pt column to be used for the particles to
+ * test against
+ * @param particle_eta The name of the eta column to be used for the particles
+ * to test against
+ * @param particle_phi The name of the phi column to be used for the particles
+ * to test against
+ * @param particle_mass The name of the mass column to be used for the particles
+ * to test against
+ * @param dR_cut The maximum dR to be used for the veto
+ * @return ROOT::RDF::RNode a dataframe containing the new veto flag
+ */
+
+ROOT::RDF::RNode DeltaRParticleVeto(
+    ROOT::RDF::RNode df, const std::string &output_flag, const std::string &p4,
+    const std::string &particle_mask, const std::string &particle_pt,
+    const std::string &particle_eta, const std::string &particle_phi,
+    const std::string &particle_mass, const float dR_cut) {
+    auto veto_overlapping_photon =
```

Comment:
Since this function is not only for photons I would rename this to `veto_overlapping` or maybe `veto_overlapping_particle`.

Reply:
done !